src/inference/src/dev/core_impl.cpp

```diff
@@ -1098,6 +1098,23 @@ std::shared_ptr<const ov::Model> ov::CoreImpl::apply_auto_batching(const std::sh
     return ov::details::apply_batch_affinity(model, deviceNameWithoutBatch);
 }
 
+void ov::CoreImpl::apply_rt_info(const std::shared_ptr<const ov::Model>& model, ov::AnyMap& config) const {
+    if (model->has_rt_info({"runtime_options", "KV_CACHE_PRECISION"})) {
```

Comment:
I am not sure that `Core` should be responsible for parsing runtime information and conversion it to properties:
1. Such properties are not supported by all plugins and e.g. for NPU it will lead to exception.
2. Here you handled only case when user passes `ov::Model` to `compile_model`, but there are also ways to pass `string + Tensor` / `filename` which is preferred way if you don't need to transform model before `compile_model` (because of model caching feature)

IMO, plugins need to take into account some Model / operations rt_info by themselves (like they currently do with primitives_priority, etc)

Reply:
We discussed this with @slyalin and @ilya-lavrenov and the decision is to read this information from the model level (as it is now) but move the implementation from Core to the plugin.
Also, the user should have the ability to override these options with config passed to .compile_model() API.

Reply:
And, btw, it would be great to have a more generic mechanism that could read options from the "<runtime_options>" sections of rt_info and use them in the runtime config of the plugin.

Reply:
Agree with the comments.

@zhangYiIntel Please move rt_into parsing in the plugin right before readProperties call (https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/intel_cpu/src/plugin.cpp#L250, https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/intel_cpu/src/plugin.cpp#L523 and https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/intel_cpu/src/plugin.cpp#L587). readProperties call is responsible to merge core config with compiled model config, so such parsing placement will allow to automatically satisfy the following requirement: 

> Also, the user should have the ability to override these options with config passed to .compile_model() API.